packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -67,17 +67,20 @@ library LibProving {
 
         // Check that the block has been proposed but has not yet been verified.
         TaikoData.SlotB memory b = state.slotB;
-        if (blockId <= b.lastVerifiedBlockId || blockId >= b.numBlocks) {
+        if (meta.id <= b.lastVerifiedBlockId || meta.id >= b.numBlocks) {
             revert L1_INVALID_BLOCK_ID();
         }
 
-        uint64 slot = blockId % config.blockRingBufferSize;
+        uint64 slot = meta.id % config.blockRingBufferSize;
         TaikoData.Block storage blk = state.blocks[slot];
 
         // Check the integrity of the block data. It's worth noting that in
         // theory, this check may be skipped, but it's included for added
         // caution.
-        if (blk.blockId != blockId || blk.metaHash != evidence.metaHash) {
+        if (
+            blk.blockId != meta.id || blk.metaHash != evidence.metaHash
```

Comment:
Evidence and Metadata are both passed to proveBlock, therefore, you should remove `metahash` from `Evidence`.